### ISSUE TITLE
Parsing timeperiods problem

### ISSUE THREAD

**Issue by petrovicboban** [pynag/pynag#243]
This is in timeperiod config file:  
```
define timeperiod{
        timeperiod_name    24x7
        alias              Always
        sunday             00:00-24:00
        monday             00:00-24:00
        tuesday            00:00-24:00
        wednesday          00:00-24:00
        thursday           00:00-24:00
        friday             00:00-24:00
        saturday           00:00-24:00
}
```
No tabs here, just regular whitespaces.    
  
  
Code below:  
```
import pprint
from pynag import Model

pp = pprint.PrettyPrinter(indent=4)
for t in Model.Timeperiod.objects.all:
    print t
```

produces:

```
define timeperiod {
  alias                          Always
  friday             00:00-24:00 
  monday             00:00-24:00 
  saturday           00:00-24:00 
  sunday             00:00-24:00 
  thursday           00:00-24:00 
  timeperiod_name                24x7
  tuesday            00:00-24:00 
  wednesday          00:00-24:00 
}
```

So, timeperiod config file is not parsed correctly where right value is non-string, which can be seen more clear using RESTloss API, which sees this timeperiod as:
```
  {
    "thursday           00:00-24:00": "", 
    "wednesday          00:00-24:00": "", 
    "friday             00:00-24:00": "", 
    "sunday             00:00-24:00": "", 
    "alias": "Always", 
    "monday             00:00-24:00": "", 
    "tuesday            00:00-24:00": "", 
    "timeperiod_name": "24x7", 
    "saturday           00:00-24:00": ""
  }
```

**Comment by pall-valmundsson** [comment]
This seems to be intentional, ref: https://github.com/pynag/pynag/commit/8b5a2f96c956ba5b34eed47226d78fc8032de0db.

@palli might have some insight.

**Comment by palli** [comment]
tl;dr Yeah this is by design (and I admit, bad design as such).

The reason for this is that for all nagios objects, except timeperiods the 'key' is unique, which is why we decided to use python dict for storing the objects. Then it later turned out that this is a perfectly valid timeperiod:

```
define timeperiod {
    thursday           01:00-02:00
    thursday           02:00-03:00
    thursday           03:00-04:00
    thursday           04:00-05:00
}
```

And same goes for exemptions. For this reason we hacked the timeperiod to keep the entire line in the key, otherwise data would be lost if pynag reads and then writes the object back to disk.

Since timeperiod does not behave like dicts, it would be wrong to try to access it as such, but the object will still behave as a dictionary for compatibility reasons.

**Comment by petrovicboban** [comment]
Thanks for explanation.
How about storing it like this:
```
{
    "thursday": ["01:00-02:00", "02:00-03:00", "03:00-04:00", "04:00-05:00"]
}
```

It's still dictionary, and it's better design.

**Comment by palli** [comment]
I think the best way forward here is to add custom helper methods in pynag.Model.Timeperiod that will make reading and editing the timeperiods easier and leave the internal data structure alone. Example:

```
>>> timeperiod =  Model.Timeperiod.objects.all[0]
>>> timeperiod.add_timerange('thursday', '01:00-02:00')
>>> timeperiod.add_timerange('thursday', '02:00-03:00')
>>> timeperiod.add_timerange('thursday', '03:00-04:00')
>>> print timeperiod.get_times()
[
    { 'day': 'thursday', 'timerange': '01:00-02:00' },
    { 'day': 'thursday', 'timerange': '02:00-03:00' },
    { 'day': 'thursday', 'timerange': '03:00-04:00' },
]
timeperiod.remove_timerange('thursday', '03:00-04:00')
```

**Comment by Crapworks** [comment]
Hi @palli,

since we are having some issues with [RESTlos](https://github.com/Crapworks/RESTlos) regarding this change (see: [Issue](https://github.com/Crapworks/RESTlos/issues/11)), I was wondering if there could be a quick workaround for that.

Long Story Short:
I am using this to get valid keys for the endpoints:
```python
        for endpoint in Model.all_attributes.object_definitions.keys():
            self[endpoint] = Model.all_attributes.object_definitions[endpoint].keys()
```

I later validate POST requests like this:
```python
            if not attr.startswith('_') and attr not in self[endpoint]:
                return {404: "unknown attribute: %s" % (attr, )}
```

This will unfortunately not work for Timeperiods, since the don't return the weekdays as keys. Of course I could use an `if` statement evereywhere to treat Timeperoids special, but I am not really looking forward to do this ;)

Any ideas for a possible abstraction layer here?

Regards,
Christian
